src/functions/pattern.ts

```diff
@@ -13,7 +13,9 @@ export interface IRulePatternOptions {
 // the available flags are "gimsuy" as described here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RegExp
 const REGEXP_PATTERN = /^\/(.+)\/([a-z]*)$/;
 
-function getFromCache(cache: Map<string, RegExp>, pattern: string): RegExp {
+const cache = new Map();
```

Comment:
I have written so many freaking time the exact same code about get or set caching that I ended up publishing https://www.npmjs.com/package/@straw-hat/fancy-map

Feel free to ignore or copy the code, or use the package 😄 just funny to see it again, and again and again LOL

Reply:
Thanks!
This cache won't stay for too long here, at least not in its current form.
I just moved it out, so that things continue to work, but I'm going to do it a bit differently.
My plan is to move it to the runtime package, buuuut prior to that I need to convert Spectral to monorepo (on our list!)